tensorboard/webapp/plugins/npmi/views/main/main_component.ts

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
+import {RunId} from '../../../../core/types';
+
+@Component({
+  selector: 'main-component',
+  templateUrl: './main_component.ng.html',
+  styleUrls: ['./main_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MainComponent {
+  @Input() runs: Map<RunId, boolean> = new Map();
+  get runActive(): boolean {
+    let active = false;
+    this.runs.forEach((runActive: boolean) => {
+      active = active || runActive;
+    });
+    return active;
```

Comment:
```ts
return this.runs.some((runActive) => runActive);
```

Reply:
Property 'some' does not exist on type 'Map<string, boolean>'.